generators/unifiedsearch-core/src/main/java/uk/gov/justice/services/clients/unifiedsearch/core/UnifiedSearchTransformerCache.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.services.clients.unifiedsearch.core;
+
+
+import uk.gov.justice.services.clients.unifiedsearch.core.domain.UnifiedSearchDescriptor;
+import uk.gov.justice.services.clients.unifiedsearch.core.parser.UnifiedSearchDescriptorFileParserFactory;
+import uk.gov.justice.services.clients.unifiedsearch.core.parser.UnifiedSearchDescriptorParser;
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.commons.collections.CollectionUtils;
+
+
+@ApplicationScoped
+public class UnifiedSearchTransformerCache {
+
+    @Inject
+    private UnifiedSearchFileFinder unifiedSearchFileFinder;
+
+    @Inject
+    private FileContentsAsStringLoader fileContentsAsStringLoader;
+
+    @Inject
+    private UnifiedSearchTransformerCacheUtils unifiedSearchTransformerCacheUtils;
+
+    private static final Object lock = new Object();
+
+    private Map<String, String> transformerCache = new ConcurrentHashMap<>();
+
+
+    public String getTransformerConfigBy(final String eventName) {
+        if (transformerCache.isEmpty()) {
+            synchronized (lock) {
+                if (transformerCache.isEmpty()) {
+                    populateCache();
+                }
+            }
+        }
+        return transformerCache.get(eventName);
+    }
+
+
+    private void populateCache() {
+        final UnifiedSearchDescriptorParser unifiedSearchDescriptorFileParser = new UnifiedSearchDescriptorFileParserFactory().create();
+        final List<Path> yamlPaths = unifiedSearchFileFinder.getUnifiedSearchDescriptor();
+        final Path basePath = unifiedSearchFileFinder.getFromClasspath("");
```

Comment:
I am not very sure about the effect of this empty string

Reply:
This is to get basePath